### PR TITLE
Proposed fix for Issue #234

### LINKED ISSUE #234: In vrf_lite, if bgp is not defined at global level, config is not rendered
https://github.com/netascode/ansible-dc-vxlan/blob/d27bdec0612521342f05f56da2c8402f133f034b/roles/dtc/common/templates/ndfc_vrf_lite.j2#L4

In vrf_lite, if `bgp` is not defined at the global level but configured at the switch level, the configuration is not rendered.

For example, this config will be rendered:
```yaml
---
vxlan:
  overlay_extensions:
    vrf_lites:
      - name: bgp_vrf_lite
        vrf: NaC-VRF02
        bgp:
          graceful_restart: true
        switches:
          - name: netascode-border-2
            bgp_peers:
              - address: 10.1.2.2
                remote_as: 65010
                description: BGP Peer 1
                address_family_ipv4_unicast:
                   default_originate: true
```

but this config will not be rendered:
```yaml
---
vxlan:
  overlay_extensions:
    vrf_lites:
      - name: bgp_vrf_lite
        vrf: NaC-VRF02
        switches:
          - name: netascode-border-2
            bgp_peers:
              - address: 10.1.2.2
                remote_as: 65010
                description: BGP Peer 1
                address_family_ipv4_unicast:
                   default_originate: true

### PR DESCRIPTION
**Pull request by GaetanoCarlucci**
## Related Issue(s)

Proposed fix to resolve #234 

## Related Collection Role

* [ ] cisco.nac_dc_vxlan.validate
* [x] cisco.nac_dc_vxlan.dtc.create
* [ ] cisco.nac_dc_vxlan.dtc.deploy
* [ ] cisco.nac_dc_vxlan.dtc.remove
* [ ] other

## Related Data Model Element

* [ ] vxlan.global
* [ ] vxlan.topology
* [ ] vxlan.underlay
* [ ] vxlan.overlay_services
* [x] vxlan.overlay_extensions
* [ ] vxlan.policy
* [ ] defaults.vxlan
* [ ] other

## Proposed Changes

Proposed fix for Issue #234

## Checklist

* [ ] Latest commit is rebased from develop with merge conflicts resolved
* [ ] New or updates to documentation has been made accordingly
* [ ] Assigned the proper reviewers

### CODE REVIEW

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+{% set bgp = {} %}
 {% if item.bgp is defined and item.bgp %}
-{% include 'ndfc_vrf_lite_ebgp.j2' %}
-{% endif %}
\ No newline at end of file
+  {% set _ = bgp.update({'enabled': true}) %}
+{% endif %}
+{% if switch_item.bgp_peers or switch_item.bgp %}
+    {% set _ = bgp.update({'enabled': true}) %}
+{% endif %}
```
Comment (devegupt) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:10]: Nit: Can we consolidate both of these if conditions into one, as both are setting the same variable/value? For example:
```
{% if (item.bgp is defined and item.bgp) or switch_item.bgp_peers or switch_item.bgp %}
  {% set _ = bgp.update({'enabled': true}) %}
{% endif %}
```

**Review by devegupt — CHANGES_REQUESTED** [review]
I have tested this code and it works, but I suggested a minor change for improving if condition.

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 [hunk 2]
```diff
@@ -1,6 +1,13 @@
 {% if item.ospf is defined and item.ospf %}
 {% include 'ndfc_vrf_lite_ospf.j2' %}
 {% endif %}
+{% set bgp = {} %}
 {% if item.bgp is defined and item.bgp %}
-{% include 'ndfc_vrf_lite_ebgp.j2' %}
-{% endif %}
\ No newline at end of file
+  {% set _ = bgp.update({'enabled': true}) %}
```
Comment (mikewiebe) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:6]: Indentation looks odd here.  Can we keep it consistent?  Looks like in most places we use 2 spaces, not 4.

Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2:6]: it was a typo, thanks for reviewing

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 (hunk 1 above)
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2:10]: i agree, thanks @devegupt !

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {% if item.ospf is defined and item.ospf %}
 {% include 'ndfc_vrf_lite_ospf.j2' %}
```
Comment (juburnet) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:2]: To keep this file consistent with spacing, we should also add 2 spaces on line 2.

**Review by juburnet — CHANGES_REQUESTED** [review]

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 (hunk 3 above)
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: thanks @juburnet ! spaces added

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {% if item.ospf is defined and item.ospf %}
-{% include 'ndfc_vrf_lite_ospf.j2' %}
+  {% include 'ndfc_vrf_lite_ospf.j2' %}
```
Comment (mtarking) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:2]: Can we change this line to:

```{% include 'ndfc_vrf_lite/ndfc_vrf_lite_ospf.j2' %}```

This means you need to create an ```ndfc_vrf_lite``` directory under ```templates``` which aligns with how we've started separating things in already committed and subsequent PRs.

Comment (mtarking) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:2]: How does the indentation of line 2 here make the rendered configuration look? Is the indentation impactful at all as we've left-aligned Jinja2 in most other files if not mistaken.

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+{% endif %}
+{% if (item.bgp is defined and item.bgp) or switch_item.bgp_peers or switch_item.bgp %}
+  {% include 'ndfc_vrf_lite_ebgp.j2' %}
```
Comment (mtarking) [inline roles/dtc/common/templates/ndfc_vrf_lite.j2:5]: Same question as from OSPF; how does the indentation of line 2 here make the rendered configuration look? Is the indentation impactful at all as we've left-aligned Jinja2 in most other files if not mistaken.

Comment (juburnet) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2:5]: @mtarking - There is no impact.  Other jinja templates have indented formatting for readability, without impact to code/cli alignment.  The jinja commands and indentation are separate from the raw config indentation (see: https://github.com/netascode/ansible-dc-vxlan/blob/develop/roles/dtc/common/templates/ndfc_vrf_lite_ospf.j2) 

Here are the results from this PR with the indented jinja template and produces proper output.

```yaml
- switch:
    - ip: 192.168.112.7
      policies:
        - create_additional_policy: False
          description: nac_myvrf_50001_vrf_lite_netascode4-ospf-bl2
          name: switch_freeform
          policy_vars:
            CONF: |-
              ! This NDFC VRF-Lite eBGP config data structure is auto-generated
              ! DO NOT EDIT MANUALLY
              !
              !
              !
              vrf context myvrf_50001
              router bgp 65003
                vrf myvrf_50001
                  bestpath as-path multipath-relax
                  graceful-restart
                  graceful-restart-helper
                  neighbor 10.31.95.95
                    remote-as 65099
                    ebgp-multihop 2
                    address-family ipv4 unicast
                      route-map extcon-rmap-filter-v6 out
                      allowas-in 3
              !
```

[the remark this answers is not in the document]
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: done, please have a look

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 (hunk 4 above)
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: done, please have a look

Comment (mtarking) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: Thanks @GaetanoCarlucci! Can you resolve the conversations as you complete them as it is a check done in GitHub that needs to be done to enable merging the PR.

Comment (mtarking) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: Thanks @juburnet! Can you resolve the conversations as you complete them as it is a check done in GitHub that needs to be done to enable merging the PR.

File: roles/dtc/common/templates/ndfc_vrf_lite.j2 (hunk 5 above)
[the remark this answers is not in the document]
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: I think that with this modification, we have introduced a conflict with the [PR](https://github.com/netascode/ansible-dc-vxlan/pull/244) on the file `ndfc_vrf_lite_ebgp.j2`

[the remark this answers is not in the document]
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: sound good, thanks @mtarking

[the remark this answers is not in the document]
Comment (GaetanoCarlucci) [reply roles/dtc/common/templates/ndfc_vrf_lite.j2]: done!

**Review by devegupt — APPROVED** [review]
This looks good to me.

**Review by juburnet — APPROVED** [review]
Reviewed indentation and file structure.

**Pull request merged**